### DISCUSSION TITLE
Template inheritance

### DISCUSSION THREAD

**Discussion by barsdeveloper** [joomla/joomla-cms#34728] [Q&A] [answered]
Hello,
I'm wondering is it possible to make template inherit from one another, to have them in a parent - child relationship? I'm interested to inherit joomla assets libraries and html overrides. I couldn't find any source of information and thought it could be usefull (for example Drupal has this feature).

**Comment by richard67** [comment]
@dgrammatiko Child templates .. your baby :-)

**Reply by barsdeveloper** [reply]
Thanks, do you have an example? Or a reference file templateDetails.xml that declares this parent child relationship.

**Comment by dgrammatiko** [comment, marked as answer]
Inheritable (child) templates are also supported in joomla 4 since https://github.com/joomla/joomla-cms/pull/30192 but the component that manages templates is not yet compatible with the new code. There’s a PR https://github.com/joomla/joomla-cms/pull/32896 for it but it’s unlikely that will be merged in time for 4.0.0.

anyways the changes that you need to have inheritable templates are:

For the devs the new parts are 2:

- manifest requires a tag <inheritable>1</inheritable > to enable a template in the new mode
- all static assets and also any static asset override should be done in the media/( administrator || site )/templateName. Folders like css/js/images inside the above path will act exactly the same as they did in the old templates/name/( css || js || images )
